### DISCUSSION TITLE
更新后无法预览 office 文件

### DISCUSSION THREAD

**Discussion by zkeq** [spencerwooo/onedrive-vercel-index#446] [Q&A] [answered]
教育版和 E5 订阅都出现了此问题。

教育版：http://tuo.icodeq.com/xiaoxuetxt

E5 :  [译林版三年级起点五年级下.xlsx](https://cdn.dream-plan.cn/zh-CN/%E5%9B%BE%E6%AC%A7%E5%A4%87%E4%BB%BD/%F0%9F%93%9C%204%23%20%E5%8D%95%E8%AF%8D%E6%96%87%E6%9C%AC/%F0%9F%93%81%201%23%20%E5%A2%A8%E5%A2%A8%E5%8D%95%E8%AF%8D/%F0%9F%93%81%204%23%20%E5%B8%A6%E8%8B%B1%E6%96%87%E9%87%8A%E4%B9%89%E7%89%88/1.%E5%85%A8%E5%9B%BD%E5%90%84%E5%A4%A7%E6%95%99%E6%9D%90%E7%89%88%E6%9C%AC%E4%B8%AD%E5%B0%8F%E5%AD%A6%E5%90%8C%E6%AD%A5/%E8%AF%91%E6%9E%97%E7%89%88/%E8%AF%91%E6%9E%97%E7%89%88%E4%B8%89%E5%B9%B4%E7%BA%A7%E8%B5%B7%E7%82%B9%E4%BA%94%E5%B9%B4%E7%BA%A7%E4%B8%8B.xlsx)

直接在官方接口后面接直链也会报错...

[https://view.officeapps.live.com/op/embed.aspx?src=后面接的是复制的直链](https://view.officeapps.live.com/op/embed.aspx?src=hhttps://cdn.dream-plan.cn/api/raw/?path=/%E5%9B%BE%E6%AC%A7%E5%A4%87%E4%BB%BD/%F0%9F%93%9C%204%23%20%E5%8D%95%E8%AF%8D%E6%96%87%E6%9C%AC/%F0%9F%93%81%201%23%20%E5%A2%A8%E5%A2%A8%E5%8D%95%E8%AF%8D/%F0%9F%93%81%204%23%20%E5%B8%A6%E8%8B%B1%E6%96%87%E9%87%8A%E4%B9%89%E7%89%88/1.%E5%85%A8%E5%9B%BD%E5%90%84%E5%A4%A7%E6%95%99%E6%9D%90%E7%89%88%E6%9C%AC%E4%B8%AD%E5%B0%8F%E5%AD%A6%E5%90%8C%E6%AD%A5/%E8%AF%91%E6%9E%97%E7%89%88/%E8%AF%91%E6%9E%97%E7%89%88%E4%B8%89%E5%B9%B4%E7%BA%A7%E8%B5%B7%E7%82%B9%E4%BA%94%E5%B9%B4%E7%BA%A7%E4%B8%8B.xlsx)

是最新版... 

![1](https://p15.qhimg.com/t019d06037d0b80338a.png)

**Comment by spencerwooo** [comment]
看来又是这个 redirect 的问题，之前是直接给官方接口提供直链，现在是用 `/api/raw` 转发了一下。

**Reply by zkeq** [reply]
我直接接直链好像也报这个错...

不知道咋了....前几天还可以明明...

https://view.officeapps.live.com/op/embed.aspx?src=https://qianrise-my.sharepoint.com/personal/tuou_exp_1ove_ml/_layouts/15/download.aspx?UniqueId=d04b4e33-4bb8-4649-a299-e6b67660546b&Translate=false&tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcWlhbnJpc2UtbXkuc2hhcmVwb2ludC5jb21AZmY2MWZkMTUtNDg5NS00MDc1LWFiMzEtOTI5NTE5NTVmMTVkIiwiaXNzIjoiMDAwMDAwMDMtMDAwMC0wZmYxLWNlMDAtMDAwMDAwMDAwMDAwIiwibmJmIjoiMTY0NDkxODU4NyIsImV4cCI6IjE2NDQ5MjIxODciLCJlbmRwb2ludHVybCI6ImMxcmJCUzVFZkp5MjR5amI4N2FXT001UFFUQkVsd1ZVbWtQMXR1Q3I3V1E9IiwiZW5kcG9pbnR1cmxMZW5ndGgiOiIxNDgiLCJpc2xvb3BiYWNrIjoiVHJ1ZSIsImNpZCI6IlpqSmhabUkxWVRJdE5UZzBaUzAwWkRJMExUazRNVFV0TWpWa1pUbGtZbVJoTkRsaiIsInZlciI6Imhhc2hlZHByb29mdG9rZW4iLCJzaXRlaWQiOiJPR0l6T0RKbVptWXRNek5pTUMwME9XVmxMV0kzT1RFdFpUQXhNV0l4TmpkbU1qUmgiLCJhcHBfZGlzcGxheW5hbWUiOiJWRVJDRUwiLCJhcHBpZCI6IjdjZjU0ZjIxLWY0YjItNDhkYS04MTMyLTA5NTQ5NmQ3NWNjMiIsInRpZCI6ImZmNjFmZDE1LTQ4OTUtNDA3NS1hYjMxLTkyOTUxOTU1ZjE1ZCIsInVwbiI6InR1b3VAZXhwLjFvdmUubWwiLCJwdWlkIjoiMTAwMzIwMDE5RTUzQjNBQSIsImNhY2hla2V5IjoiMGguZnxtZW1iZXJzaGlwfDEwMDMyMDAxOWU1M2IzYWFAbGl2ZS5jb20iLCJzY3AiOiJhbGxmaWxlcy5yZWFkIGFsbGZpbGVzLndyaXRlIGFsbHByb2ZpbGVzLnJlYWQiLCJ0dCI6IjIiLCJ1c2VQZXJzaXN0ZW50Q29va2llIjpudWxsLCJpcGFkZHIiOiIyMC4xOTAuMTUxLjE2MCJ9.dGgxcllTMWk4emhNVklyTmR6ay9PYitXRzh0b1JkSUxIaGY3YTlHRWIxTT0&ApiVersion=2.0

https://view.officeapps.live.com/op/embed.aspx?src=https://zkeq-my.sharepoint.com/personal/zkeq_zkeq_onmicrosoft_com/_layouts/15/download.aspx?UniqueId=e526d5e2-67b7-43cc-a7fe-88559589f8bb&Translate=false&tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvemtlcS1teS5zaGFyZXBvaW50LmNvbUA4MDllZjIxNi0wMTA5LTRhY2QtYTAxZi0xZTBlOTMxNzg4NDYiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAiLCJuYmYiOiIxNjQ0OTE4NzU5IiwiZXhwIjoiMTY0NDkyMjM1OSIsImVuZHBvaW50dXJsIjoiL2E2bFQva3krZnltbFpWTEp5NzVUamRDUFFjczA3V1V0OEluQk9DdjF1QT0iLCJlbmRwb2ludHVybExlbmd0aCI6IjE1MyIsImlzbG9vcGJhY2siOiJUcnVlIiwiY2lkIjoiTW1Ka09HSTNZamN0TkdOaFpTMDBaREEyTFdKbVpXVXRaVGd6TVdKaVkySXhZbUkyIiwidmVyIjoiaGFzaGVkcHJvb2Z0b2tlbiIsInNpdGVpZCI6Ik5tWmlaakZrWWpjdFl6UTVNUzAwTlRoaExUZzBPR1V0WXpGaE1UYzVabU0yTjJWbSIsImFwcF9kaXNwbGF5bmFtZSI6Im9uZWRyaXZlLXZlcmNlbC1pbmRleCIsImdpdmVuX25hbWUiOiJrZXEiLCJmYW1pbHlfbmFtZSI6InoiLCJzaWduaW5fc3RhdGUiOiJbXCJrbXNpXCJdIiwiYXBwaWQiOiJkODdiY2MzOS0xNzUwLTRjYTAtYWQ1NC1mOGQwZWZiYjI3MzUiLCJ0aWQiOiI4MDllZjIxNi0wMTA5LTRhY2QtYTAxZi0xZTBlOTMxNzg4NDYiLCJ1cG4iOiJ6a2VxQHprZXEub25taWNyb3NvZnQuY29tIiwicHVpZCI6IjEwMDMyMDAxRDRGOTlBMTciLCJjYWNoZWtleSI6IjBoLmZ8bWVtYmVyc2hpcHwxMDAzMjAwMWQ0Zjk5YTE3QGxpdmUuY29tIiwic2NwIjoiYWxsZmlsZXMucmVhZCBhbGxwcm9maWxlcy5yZWFkIiwidHQiOiIyIiwidXNlUGVyc2lzdGVudENvb2tpZSI6bnVsbCwiaXBhZGRyIjoiMjAuMTkwLjE1MS4zNyJ9.eEV6NzhDaFNHUldraUdOOFhhcjFpK0h6cmNmdHJmVVkzeEtuNmIzZm5KOD0&ApiVersion=2.0

**Reply by zkeq** [reply]
哦哦之前的 url 是编码后的，而且是直链，可以正常访问..

https://view.officeapps.live.com/op/embed.aspx?src=https%3A%2F%2Fqianrise-my.sharepoint.com%2Fpersonal%2Ftuou_exp_1ove_ml%2F_layouts%2F15%2Fdownload.aspx%3FUniqueId%3Dd04b4e33-4bb8-4649-a299-e6b67660546b%26Translate%3Dfalse%26tempauth%3DeyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcWlhbnJpc2UtbXkuc2hhcmVwb2ludC5jb21AZmY2MWZkMTUtNDg5NS00MDc1LWFiMzEtOTI5NTE5NTVmMTVkIiwiaXNzIjoiMDAwMDAwMDMtMDAwMC0wZmYxLWNlMDAtMDAwMDAwMDAwMDAwIiwibmJmIjoiMTY0NDkxODU4NyIsImV4cCI6IjE2NDQ5MjIxODciLCJlbmRwb2ludHVybCI6ImMxcmJCUzVFZkp5MjR5amI4N2FXT001UFFUQkVsd1ZVbWtQMXR1Q3I3V1E9IiwiZW5kcG9pbnR1cmxMZW5ndGgiOiIxNDgiLCJpc2xvb3BiYWNrIjoiVHJ1ZSIsImNpZCI6IlpqSmhabUkxWVRJdE5UZzBaUzAwWkRJMExUazRNVFV0TWpWa1pUbGtZbVJoTkRsaiIsInZlciI6Imhhc2hlZHByb29mdG9rZW4iLCJzaXRlaWQiOiJPR0l6T0RKbVptWXRNek5pTUMwME9XVmxMV0kzT1RFdFpUQXhNV0l4TmpkbU1qUmgiLCJhcHBfZGlzcGxheW5hbWUiOiJWRVJDRUwiLCJhcHBpZCI6IjdjZjU0ZjIxLWY0YjItNDhkYS04MTMyLTA5NTQ5NmQ3NWNjMiIsInRpZCI6ImZmNjFmZDE1LTQ4OTUtNDA3NS1hYjMxLTkyOTUxOTU1ZjE1ZCIsInVwbiI6InR1b3VAZXhwLjFvdmUubWwiLCJwdWlkIjoiMTAwMzIwMDE5RTUzQjNBQSIsImNhY2hla2V5IjoiMGguZnxtZW1iZXJzaGlwfDEwMDMyMDAxOWU1M2IzYWFAbGl2ZS5jb20iLCJzY3AiOiJhbGxmaWxlcy5yZWFkIGFsbGZpbGVzLndyaXRlIGFsbHByb2ZpbGVzLnJlYWQiLCJ0dCI6IjIiLCJ1c2VQZXJzaXN0ZW50Q29va2llIjpudWxsLCJpcGFkZHIiOiIyMC4xOTAuMTUxLjE2MCJ9.dGgxcllTMWk4emhNVklyTmR6ay9PYitXRzh0b1JkSUxIaGY3YTlHRWIxTT0%26ApiVersion%3D2.0

**Comment by spencerwooo** [comment]
其实我觉得甚至都不用 `preview-office-docs`，可以考虑：https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_preview?view=odsp-graph-online

**Reply by zkeq** [reply]
嗯嗯，我先用昨天的版本..

**Comment by spencerwooo** [comment, marked as answer]
我看我的 demo 站点能正常访问的呀？比如：

- https://drive.swo.moe/Lecture%20and%20Coursework%20CS%20(BIT)/2018%20-%20%E5%A4%A7%E4%BA%8C%E4%B8%8B%20-%20%E8%AE%A1%E7%AE%97%E6%9C%BA%E7%AE%97%E6%B3%95%E8%AE%BE%E8%AE%A1%E4%B8%8E%E5%88%86%E6%9E%90/%E8%AE%A1%E7%AE%97%E7%90%86%E8%AE%BA_%E5%8F%AF%E8%AE%A1%E7%AE%97%E7%90%86%E8%AE%BA2018.pptx
- https://drive.swo.moe/Lecture%20and%20Coursework%20CS%20(BIT)/2018%20-%20%E5%A4%A7%E4%BA%8C%E4%B8%8B%20-%20%E8%AE%A1%E7%AE%97%E6%9C%BA%E7%AE%97%E6%B3%95%E8%AE%BE%E8%AE%A1%E4%B8%8E%E5%88%86%E6%9E%90/%E8%80%83%E8%AF%95%E6%A0%B7%E9%A2%98.docx

**Reply by spencerwooo** [reply]
`<iframe>` 里面的 URL 是：

```
https://view.officeapps.live.com/op/embed.aspx?src=https://drive.swo.moe/api/raw/?path=/Lecture%20and%20Coursework%20CS%20(BIT)/2018%20-%20%E5%A4%A7%E4%BA%8C%E4%B8%8B%20-%20%E8%AE%A1%E7%AE%97%E6%9C%BA%E7%AE%97%E6%B3%95%E8%AE%BE%E8%AE%A1%E4%B8%8E%E5%88%86%E6%9E%90/%E8%80%83%E8%AF%95%E6%A0%B7%E9%A2%98.docx
```

**Reply by zkeq** [reply]
确实，但是换成我部署的直链就不行...

不知道什么原因

```
https://view.officeapps.live.com/op/embed.aspx?src=https://tuostudy.vercel.app/api?path=/📜%20050%23%20单词文本/📁%2001%23%20墨墨单词（超全）/📁%2003%23%20带中文释义版/1.全国各大教材版本中小学同步/人教版/人教版一年级起点三年级上.xlsx&raw=true
```

```
https://view.officeapps.live.com/op/embed.aspx?src=https://cdn.dream-plan.cn/api/raw/?path=/图欧备份/📜%204%23%20单词文本/📁%201%23%20墨墨单词/📁%203%23%20带中文释义版/9.其他（更多）/英语词组全书（上）.xlsx
```

**Reply by spencerwooo** [reply]
不会是因为 emoji 没 urlEncode 吧？

**Reply by zkeq** [reply]
> 不会是因为 emoji 没 urlEncode 吧？

好像就是这个原因，我传到了根目录一份可以正常加载....

```
https://view.officeapps.live.com/op/embed.aspx?src=https://cdn.dream-plan.cn/api/raw/?path=/%E8%8B%B1%E8%AF%AD%E8%AF%8D%E7%BB%84%E5%85%A8%E4%B9%A6%EF%BC%88%E4%B8%8A%EF%BC%89.xlsx
```
